test/Generators.hs

```diff
@@ -15,17 +15,48 @@ import qualified Hedgehog.Range as Range
 
 -- DISCREPANCY: 'jq' allows both 000 in input and its own number literals,
 -- but JSON spec does not. We'll stay closest to 'jq' by doing this, too.
+--
 
--- First, an unsized Gen of JSON numbers.
+-- | Helper Generators
+signGen :: Gen Text
+signGen = Gen.element ["+", "-", ""]
 
+signGen' :: Gen Text
+signGen' = Gen.element ["-", ""]
+
+integerGen :: Gen Text
+integerGen = Text.pack . show <$>
+      (Gen.integral $ Range.exponential 1 (10^309))
+
+expGen :: Gen Text
+expGen = Text.pack . show <$>
+  (Gen.integral (Range.linearFrom 0 (-10^16) (10^16)))
```

Comment:
This range generates an integer to go in the exponent place `base ^ exponent`, it doesn't actually generate the exponent itself. That is why `10^16`, `10e16` isn't valid haskell.

Additionally, I am using linearFrom because you mentioned that you wanted the exponent value to scale linearly
>But for the "exponent" part of a Scientific, we probably want to grow it linearBounded (proportional to the internal size parameter) by the total range of Scientific's base10Exponent :: Int.